beater/config/java_attacher.go

```diff
@@ -0,0 +1,41 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package config
+
+// JavaAttacherConfig holds configuration information for running a java
+// attacher jarfile.
+type JavaAttacherConfig struct {
+	Enabled bool `config:"enabled"`
+
+	IncludeAll bool     `config:"include_all"`
+	IncludePID []string `config:"include_pid"`
+
+	IncludeMain  []string `config:"include_main"`
```

Comment:
A note for the implementation: All the values will have to be escaped when passing them to the attacher to avoid injection attack vectors.

Example:
```
 - include_main: "foo; rm -rf /"
```

should not lead to the hard drive being purged

Reply:
:100: